src/averageAges.js

```diff
@@ -38,6 +52,33 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+
+  let totalAge = 0;
+  let average = 0;
+
+  const females = people.filter(human => human.sex === 'f');
+
+  const mothersList = [];
+
+  for (const human of people) {
+    mothersList.push(human.mother);
+  }
+
+  const femalesWithChildren = females.filter(human =>
+    mothersList.some(mother => mother === human.name));
+
+  let womens;
+
+  typeof withChildren === 'undefined'
+    ? womens = females
+    : womens = femalesWithChildren;
+
+  totalAge = womens.map(human =>
+    human.died - human.born).reduce((a, b) => (a + b), 0);
+
+  average = totalAge / womens.length;
+
+  return average;
```

Comment:
```suggestion
return totalAge / womens.length;
```
And the same in the rest functions